apps/ocean-api/src/index.ts

```diff
@@ -1,56 +1,32 @@
 import { RootModule } from './modules/RootModule'
-import { FastifyAdapter, NestFastifyApplication } from '@nestjs/platform-fastify'
+import { NestFastifyApplication } from '@nestjs/platform-fastify'
 import { NestFactory } from '@nestjs/core'
 import { ConfigService } from '@nestjs/config'
+import { RootServer } from '@defichain-apps/libs/rootserver'
 
-export class RootServer {
-  app?: NestFastifyApplication
-
+export class OceanApiServer extends RootServer {
   async create (): Promise<NestFastifyApplication> {
-    const adapter = new FastifyAdapter()
-    return await NestFactory.create<NestFastifyApplication>(RootModule, adapter)
+    return await NestFactory.create<NestFastifyApplication>(RootModule, this.adapter)
   }
 
   async configure (app: NestFastifyApplication, config: ConfigService): Promise<void> {
+    await super.configure(app, config)
+
     const version = config.get<string>('API_VERSION') as string
     const network = config.get<string>('API_NETWORK') as string
```

Comment:
I think is okay to include this to abstract layer too, also `setGlobalPrefix` if version and network envvar available.

Reply:
i implemented this but find out not so suitable :sweat: